src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/EditDocsAboutToSubmitHandler.java

```diff
@@ -0,0 +1,129 @@
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class EditDocsAboutToSubmitHandler implements PreSubmitCallbackHandler<AsylumCase> {
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+        return callback.getEvent() == Event.EDIT_DOCS && callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage,
+                                                        Callback<AsylumCase> callback) {
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+        restoreDocumentTagForDocs(callback, asylumCase);
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    private void restoreDocumentTagForDocs(Callback<AsylumCase> callback, AsylumCase asylumCase) {
+        getFieldDefAndFallbackDocumentTagPairs().forEach(
+            pair -> restoreDocumentTagValue(callback, asylumCase, pair.getKey(), pair.getValue()));
+    }
+
+    private void restoreDocumentTagValue(Callback<AsylumCase> callback, AsylumCase asylumCase,
+                                         AsylumCaseFieldDefinition fieldDefinition, DocumentTag fallbackDocumentTag) {
+        Optional<List<IdValue<DocumentWithMetadata>>> idValuesOptional = asylumCase.read(fieldDefinition);
+        idValuesOptional.ifPresent(idValues -> {
+            List<IdValue<DocumentWithMetadata>> idValuesBefore = getIdValuesBefore(callback, fieldDefinition);
+            asylumCase.write(fieldDefinition,
+                getIdValuesWithBeforeDocumentTags(idValues, idValuesBefore, fallbackDocumentTag));
+        });
+    }
+
+    private List<AbstractMap.SimpleImmutableEntry<AsylumCaseFieldDefinition, DocumentTag>> getFieldDefAndFallbackDocumentTagPairs() {
+        return Arrays.asList(
+            new AbstractMap.SimpleImmutableEntry<>(ADDITIONAL_EVIDENCE_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(TRIBUNAL_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(HEARING_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(LEGAL_REPRESENTATIVE_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(ADDENDUM_EVIDENCE_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(RESPONDENT_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(DRAFT_DECISION_AND_REASONS_DOCUMENTS, NONE),
+            new AbstractMap.SimpleImmutableEntry<>(FINAL_DECISION_AND_REASONS_DOCUMENTS, NONE)
```

Comment:
as everything defaulting to NONE, do we need this List of Maps?

Reply:
@makowalski , I intentionally left it NONE by default because the ticket is not clear about this scenario. However, I am going to put a comment in the ticket for our BA to clarify it. If it's all NONE  yeah, no need to have it, agree. comment added here; https://tools.hmcts.net/jira/browse/RIA-2783